pkg/util/node/node.go

```diff
@@ -182,7 +182,7 @@ func PatchNodeStatus(c v1core.CoreV1Interface, nodeName types.NodeName, oldNode
 		return nil, nil, err
 	}
 
-	updatedNode, err := c.Nodes().Patch(string(nodeName), types.StrategicMergePatchType, patchBytes, "status")
+	updatedNode, err := c.Nodes().Patch(string(nodeName), types.MergePatchType, patchBytes, "status")
```

Comment:
the main thing I'm concerned about with this change is conditions, which also had a merge patch strategy, and is the most likely of the status fields to be written by multiple clients concurrently.

Reply:
Hm... yeah.

So, we could do a strategic merge of everything except the `Addresses` field, and then (if `Addresses` changed) do a second non-strategic merge of just that?

Reply:
@danwinship theres only like 2 or 3 places that Addresses should change. So perhaps we create a separate PatchNodeStatusAddresses() and use that from those 3 places instead?

Reply:
Interesting - it looks like SetNodeConditions doesn't *really* patch... is that OK?

Reply:
I think this last is the sanest approach - change as little as possible.